### PR TITLE
enhancement(influxdb sink, codecs): Integrate `encoding::Encoder` with `influxdb` sink

### PR DESCRIPTION
**Pull request by pablosichert**
Part of #9459.

Technically, this doesn't use `Encoder`, but only the new simplified `Transformer`.

### CODE REVIEW

File: src/sinks/influxdb/logs.rs [hunk 1]
```diff
@@ -174,14 +166,16 @@ struct InfluxDbLogsEncoder {
...
 }
 
 impl HttpEventEncoder<BytesMut> for InfluxDbLogsEncoder {
     fn encode_event(&mut self, event: Event) -> Option<BytesMut> {
         let mut event = event.into_log();
         event.insert("metric_type", "logs".to_string());
-        self.encoding.apply_rules(&mut event);
+        let mut event = Event::from(event);
+        self.transformer.transform(&mut event);
+        let mut event = event.into_log();
```
Comment (jszwedko) [inline src/sinks/influxdb/logs.rs:178]:
```suggestion
        let mut event = {
          let mut event = Event::from(event);
          self.transformer.transform(&mut event);
          event.into_log()
      }
```

Maybe only marginally more readable. I just found what is here with the reassignment to `event` a bit hard to follow.

Comment (pablosichert) [reply src/sinks/influxdb/logs.rs]: Did a change in the same spirit: 3d4d34dc215cf92a3e29668729126ccb641ef3da

**Review by lukesteensen — APPROVED** [review]
